javers-core/src/main/java/org/javers/common/reflection/ReflectionUtil.java

```diff
@@ -62,25 +60,22 @@ public static Object invokeGetter(Object target, String getterName) {
 
     /**
      * Creates new instance of public or package-private class.
-     * Calls first, not-private constructor
      */
     public static Object newInstance(Class clazz, ArgumentResolver resolver){
         Validate.argumentIsNotNull(clazz);
+        JaversException ex = null;
         for (Constructor constructor : clazz.getDeclaredConstructors()) {
             if (isPrivate(constructor) || isProtected(constructor)) {
                 continue;
             }
 
             Class [] types = constructor.getParameterTypes();
             Object[] params = new Object[types.length];
-            for (int i=0; i<types.length; i++){
-                try {
-                    params[i] = resolver.resolve(types[i]);
-                } catch (JaversException e){
-                    logger.error("failed to create new instance of "+clazz.getName()+", argument resolver for arg["+i+"] " +
-                                 types[i].getName() + " thrown exception: "+e.getMessage());
-                    throw e;
-                }
+            try {
```

Comment:
to be honest, i dont know what did change in logic of this method (besides refactoring),
could you please explain the rationale behind this change?

Reply:
I moved the loop that resolved the constructor's parameters into a separate method. The main idea was that, if the constructor parameters resolving fails, the exception is stored in the `ex` variable (https://github.com/javers/javers/pull/683/commits/0f43b6cd37f170f8727a9f2ddc69b0773ae32950#diff-1e490987d4273f55c02bf8d2abf7a6adR66) and next public constructor is then tried. Should other constructors also fail, the exception is thrown at the end. If any of the other constructors succeed in instantiating the class object then the exception won't be thrown.